### PR TITLE
Bug fix: tf.assert_* prints the entire input by default in eager mode

### PR DESCRIPTION
**Pull request by frreiss**
With the exception of `assert_equal`, all the ops that invoke `tf.Assert` will print out their input tensors in their entirety by default when running in eager mode. This behavior is different from the behavior when running in graph mode, and it's also somewhat dangerous. For example,
```
tf.assert_greater(np.zeros([1000,1000], dtype=np.float32), 0.)
```
will generate a five megabyte error message when running in eager mode.

This pull request changes the behavior of the `summarize` argument of `tf.Assert` (and by extension all the other ops that call `tf.Assert) to be the same in eager mode as in graph mode. Specifically, after the changes in this PR, the default number of elements to print will be 3, and values of `summarize` less than zero will cause all elements to be printed. I also added a test case.

### CODE REVIEW

**Review by rmlarsen — DISMISSED** [review]

File: tensorflow/python/kernel_tests/check_ops_test.py [hunk 1]
```diff
@@ -302,6 +302,44 @@ def test_returns_none_with_eager(self):
       x = check_ops.assert_none_equal(t1, t2)
       assert x is None
 
+  def test_error_message_eager(self):
```
Comment (alextp) [inline tensorflow/python/kernel_tests/check_ops_test.py]: Can you make the test less strict? I foresee many many changes to this test as small details of the error messasge might change in the future.

How about just checking for the presence of the tensor contents?

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/kernel_tests/check_ops_test.py (hunk 1 above)
Comment (frreiss) [reply tensorflow/python/kernel_tests/check_ops_test.py]: Sure, I'd be happy to. Will push another commit tomorrow.

**Comment by rmlarsen** [comment]
@alextp can you take another look, please?

**Review by alextp — APPROVED** [review]

File: tensorflow/python/kernel_tests/check_ops_test.py (hunk 1 above)
Comment (frreiss) [reply tensorflow/python/kernel_tests/check_ops_test.py]: Changes pushed. Regression tests pass on my test machine.

**Review by alextp — APPROVED** [review]

**Pull request merged**